testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/theme/ExtendingThemeTest.java

```diff
@@ -0,0 +1,69 @@
+package org.keycloak.testsuite.theme;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.testsuite.AbstractKeycloakTest;
+import org.keycloak.theme.Theme;
+import org.keycloak.theme.ThemeProvider;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:vincent.letarouilly@gmail.com">Vincent Letarouilly</a>
+ */
+public class ExtendingThemeTest extends AbstractKeycloakTest {
+
+    private static final String THEME_NAME = "environment-agnostic";
+
+    @Before
+    public void setUp() {
+        System.setProperty("existing_system_property", "Keycloak is awesome");
+    }
+
+    @Override
+    public void addTestRealms(List<RealmRepresentation> testRealms) {
+    }
+
+    // KEYCLOAK-6698
+    @Test
+    public void systemPropertiesSubstitutionInThemeProperties() {
+        testingClient.server().run(session -> {
+            try {
+                ThemeProvider extending = session.getProvider(ThemeProvider.class, "extending");
+                Theme theme = extending.getTheme(THEME_NAME, Theme.Type.LOGIN);
+                Assert.assertEquals("Keycloak is awesome", theme.getProperties().getProperty("system.property.found"));
+                Assert.assertEquals("", theme.getProperties().getProperty("system.property.missing"));
+                Assert.assertEquals("defaultValue", theme.getProperties().getProperty("system.property.missing.with.default"));
+            } catch (IOException e) {
+                Assert.fail(e.getMessage());
+            }
+        });
+    }
+
+    // KEYCLOAK-6698
+    @Test
+    public void environmentVariablesSubstitutionInThemeProperties() {
+        testingClient.server().run(session -> {
+            try {
+                ThemeProvider extending = session.getProvider(ThemeProvider.class, "extending");
+                Theme theme = extending.getTheme(THEME_NAME, Theme.Type.LOGIN);
+                Assert.assertEquals("", theme.getProperties().getProperty("env.missing"));
+                Assert.assertEquals("defaultValue", theme.getProperties().getProperty("env.missingWithDefault"));
+                if (System.getenv().containsKey("HOMEPATH")) {
+                    // Windows
+                    Assert.assertEquals(System.getenv().get("HOMEPATH"), theme.getProperties().getProperty("env.windowsHome"));
+                } else if (System.getenv().containsKey("HOME")) {
```

Comment:
Need to confirm if this is set on Mac

Reply:
`$HOME` seems to be ok for most of the other OS https://en.wikipedia.org/wiki/Home_directory
There's no official doc for mac, but some of my coworkers uses macs and it is set.